platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformCoordinator.java

```diff
@@ -0,0 +1,88 @@
+    private final LinkedEventIntakeWiring linkedEventIntakeWiring;
+
+    /**
+     * Constructor
+     *
+     * @param internalEventValidatorWiring  the internal event validator wiring
+     * @param eventDeduplicatorWiring       the event deduplicator wiring
+     * @param eventSignatureValidatorWiring the event signature validator wiring
+     * @param orphanBufferWiring            the orphan buffer wiring
+     * @param inOrderLinkerWiring           the in order linker wiring
+     * @param linkedEventIntakeWiring       the linked event intake wiring
+     */
+    public PlatformCoordinator(
+            @NonNull final InternalEventValidatorWiring internalEventValidatorWiring,
+            @NonNull final EventDeduplicatorWiring eventDeduplicatorWiring,
+            @NonNull final EventSignatureValidatorWiring eventSignatureValidatorWiring,
+            @NonNull final OrphanBufferWiring orphanBufferWiring,
+            @NonNull final InOrderLinkerWiring inOrderLinkerWiring,
+            @NonNull final LinkedEventIntakeWiring linkedEventIntakeWiring) {
+
+        this.internalEventValidatorWiring = Objects.requireNonNull(internalEventValidatorWiring);
+        this.eventDeduplicatorWiring = Objects.requireNonNull(eventDeduplicatorWiring);
+        this.eventSignatureValidatorWiring = Objects.requireNonNull(eventSignatureValidatorWiring);
+        this.orphanBufferWiring = Objects.requireNonNull(orphanBufferWiring);
+        this.inOrderLinkerWiring = Objects.requireNonNull(inOrderLinkerWiring);
+        this.linkedEventIntakeWiring = Objects.requireNonNull(linkedEventIntakeWiring);
+    }
+
+    /**
+     * Safely clears the intake pipeline
+     * <p>
+     * Future work: this method should be expanded to coordinate the clearing of the entire system
+     */
+    public void clear() {
+        // pause the linked event intake, to prevent any new events from making it through the intake pipeline
+        linkedEventIntakeWiring.pauseInput().inject(true);
+        linkedEventIntakeWiring.flushRunnable().run();
+
+        // flush everything remaining in the intake pipeline out into the void
+        internalEventValidatorWiring.flushRunnable().run();
+        eventDeduplicatorWiring.flushRunnable().run();
+        eventSignatureValidatorWiring.flushRunnable().run();
+        orphanBufferWiring.flushRunnable().run();
+        inOrderLinkerWiring.flushRunnable().run();
+        linkedEventIntakeWiring.flushRunnable().run();
+
+        // once everything has been flushed out of the system, it's safe to unpause the linked event intake
+        linkedEventIntakeWiring.pauseInput().inject(false);
+
+        // data is no longer moving through the system. clear all the internal data structures in the wiring objects
+        eventDeduplicatorWiring.clearInput().inject(new ClearTrigger());
+        eventDeduplicatorWiring.flushRunnable().run();
+        orphanBufferWiring.clearInput().inject(new ClearTrigger());
```

Comment:
Is there a reason the ClearTrigger is created multiple times as opposed to being a static singleton?